### PR TITLE
Relax esbuild semver dependency

### PR DESCRIPTION
**Pull request by peschee**
## What I did

1. Changed the semver from `^0.12.21` to `^0.12 || ^0.13 || ^0.14`.

I'm not sure how to check for all possible incompatibilities, but all tests still run when checking with the latest `0.12.x`, `0.13.x` and `0.14.x` version.

### CODE REVIEW

File: packages/dev-server-esbuild/package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
   "dependencies": {
     "@mdn/browser-compat-data": "^4.0.0",
     "@web/dev-server-core": "^0.3.17",
-    "esbuild": "^0.12.21",
+    "esbuild": "^0.12|^0.13|^0.14",
```
Comment (Westbrook) [inline packages/dev-server-esbuild/package.json:55]: Is `0.14` “latest”? Would we benefit from `>0.12` to simplify in the future?

Comment (peschee) [reply packages/dev-server-esbuild/package.json:55]: yes, `0.14` is latest. having the `|` in there with 12, 13 and 14 is more restrictive than `>0.12`. I can live with both.

Comment (abdonrd) [inline packages/dev-server-esbuild/package.json:55]: I think this should be like this:

```suggestion
    "esbuild": "^0.12 || ^0.13 || ^0.14",
```

Comment (abdonrd) [reply packages/dev-server-esbuild/package.json:55]: But I this works better:

```suggestion
    "esbuild": ">=0.12",
```

(but maybe we should update it in a future breaking changes)

Comment (peschee) [reply packages/dev-server-esbuild/package.json]: Yes ofc it should be `||`… I have changed that.

I can also change it to `>=0.12` if everybody is ok with that.

**Pull request merged**